DOC-12210: 3.1.8 Release notes

ElliotFrancisHunter opened the pull request:
Release notes for maintenance release 3.1.8

Changelist:
-- Release notes for Android platform
-- Moving 'What's new in 3.1' NOTE admonition to the foot of each platforms release notes
-- Updating antora.yml to reflect new maintenance version
-- Fly by fix to remove dead pages

ElliotFrancisHunter committed:
DOC-12210: 3.1.8 Release notes

Changelist:
-- Release notes for all platforms
-- Updating C downloads list with latest platform
-- Updating antora.yml to reflect new maintenance version

bfavini commented on modules/android/partials/release-notes/couchbase-mobile-android-release-note.3.1.8.adoc:
```diff
@@ -0,0 +1,21 @@
+
+Version 3.1.8 for {param-title} delivers the following features and enhancements:
+
+NOTE: For an overview of the latest features offered in Couchbase Lite 3.1, see xref:ROOT:cbl-whatsnew.adoc[New in 3.1]
```
Consider moving this note...it's a little jarring to see it right after your intro sentence.

ElliotFrancisHunter replied:
Moved it to the foot of release notes and applied this change to all of them across the platforms. Looks much better thank you! 👍🏼

bfavini commented on modules/android/partials/release-notes/couchbase-mobile-android-release-note.3.1.8.adoc:
```diff
@@ -0,0 +1,21 @@
+
+=== Issues and Resolutions
+
+* https://issues.couchbase.com/browse/CBL-5723[CBL-5723 -- Fixed Replicator syncs from beginning when using prebuilt dbs synced from SG]
```
"Fixed replicator syncs from beginning" seems awkward and unclear. After looking at the ticket, I'm thinking something like this might be clearer:

Ensured replicator syncs do not restart from the beginning when using prebuilt databases synced from Sync Gatweay.

But please check this with your SMEs.

ElliotFrancisHunter replied:
Done!

bfavini reviewed (changes requested):
A couple of small suggestions...please let me know if you have any questions.

ElliotFrancisHunter committed:
DOC-12210: Platform and android note fix

Change list:

-- Only Android platform is being updated so removing other platform release notes
-- Updated copy after feedback on android release note

ElliotFrancisHunter committed:
DOC-12210: Dead page removal.

3.0.x pages should be removed for versions 3.1.x onwards

ElliotFrancisHunter committed:
DOC-12210: Feedback and formatting fixes

Change list:
-- Tidying document spacing for consistency across all release notes
-- The NOTE admonitions former position was said to be jarring in feedback, I agree. Moved to almost serve as the footer of each release note.
--- Applied to every version on every platform for consistency

ElliotFrancisHunter committed:
DOC-12110: Fix platform versioning
